src/control/drpc/module_svc_test.go

```diff
@@ -0,0 +1,252 @@
+// portions thereof marked with this legend must also reproduce the markings.
+
+package drpc
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+const defaultTestModID int32 = 7
+
+// mockModule is a test module used for verifying the drpc.ModuleService
+type mockModule struct {
+	HandleCallResponse []byte
+	HandleCallErr      error
+	IDValue            int32
+}
+
+func (m *mockModule) HandleCall(client *Client, method int32, input []byte) ([]byte, error) {
+	return m.HandleCallResponse, m.HandleCallErr
+}
+
+func (m *mockModule) InitModule(state ModuleState) {}
+
+func (m *mockModule) ID() int32 {
+	return m.IDValue
+}
+
+func newTestModule(ID int32) *mockModule {
+	return &mockModule{
+		IDValue: ID,
+	}
+}
+
+func TestNewModuleService(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	service := NewModuleService(log)
+
+	if service == nil {
+		t.Fatal("service was nil")
+	}
+
+	common.AssertEqual(t, len(service.modules), 0, "expected empty module list")
+}
+
+func TestService_RegisterModule_Single_Success(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	service := NewModuleService(log)
+	expectedID := defaultTestModID
+	testMod := newTestModule(expectedID)
+
+	err := service.RegisterModule(testMod)
+
+	if err != nil {
+		t.Fatalf("expected no error, got: %v", err)
+	}
+
+	mod, ok := service.modules[expectedID]
+
+	if !ok {
+		t.Fatalf("module wasn't found under ID %d", expectedID)
+	}
+
+	if diff := cmp.Diff(testMod, mod); diff != "" {
+		t.Fatalf("(-want, +got)\n%s", diff)
+	}
+}
+
+func TestService_RegisterModule_Multiple_Success(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	service := NewModuleService(log)
+	expectedIDs := []int32{-1, 2, 255, defaultTestModID}
+	testMods := make([]*mockModule, 0, len(expectedIDs))
+
+	for _, id := range expectedIDs {
+		mod := newTestModule(id)
+		testMods = append(testMods, mod)
+
+		err := service.RegisterModule(mod)
+
+		if err != nil {
+			t.Fatalf("expected no error, got: %v", err)
+		}
+	}
+
+	for i, id := range expectedIDs {
+		mod, ok := service.modules[id]
+
+		if !ok {
+			t.Fatalf("registered module %d wasn't found", id)
+		}
+
+		if diff := cmp.Diff(testMods[i], mod); diff != "" {
+			t.Fatalf("(-want, +got)\n%s", diff)
+		}
+	}
+}
+
+func TestService_RegisterModule_DuplicateID(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	service := NewModuleService(log)
+	testMod := newTestModule(15)
+	dupMod := newTestModule(testMod.IDValue)
+
+	err := service.RegisterModule(testMod)
+	if err != nil {
+		t.Fatalf("expected no error for initial registration, got: %v",
+			err)
```

Comment:
Fixed. I thought perhaps when it doesn't affect readability it might still make sense to follow our C width limit. I agree this looks better on the same line though.